script.js

```diff
@@ -0,0 +1,168 @@
+// DOM 조작(검색)
+const form = document.querySelector('#input-form');
+const input = document.querySelector('#input-form input');
+const yetList = document.querySelector('#yet-list');
+const doneList = document.querySelector('#done-list');
+const todoNum = document.querySelector('#todoNum');
+const doneNum = document.querySelector('#doneNum');
+
+// 텍스트는 변수로 설정 => 오타 방지 및 수정 용이
+const yetText = 'yetList';
+const doneText = 'doneList';
+const yetBtnText = 'yetBtn';
+
+// for localStorage
+let yetArr = [];
+let doneArr = [];
+let id = 0;
+
+// 사용한 함수들에 대한 기능명세는 다음과 같음.
+// 함수1, setValue in localStorage
+const setYetLocal = (arr) => {
+  localStorage.setItem(yetText, JSON.stringify(arr));
+};
+// 함수2, setValue in localStorage
+const setDoneLocal = (arr) => {
+  localStorage.setItem(doneText, JSON.stringify(arr));
+};
+
+// 함수3, render in yetList
+const setYet = (obj) => {
+  // DOM 조작(생성)
+  const li = document.createElement('li');
+  const span = document.createElement('span');
+  const delImg = document.createElement('img');
+  const yetBtn = document.createElement('button');
+  // span.innerText = obj.contents;
+  delImg.src = './img/bin.png';
+  yetBtn.className = yetBtnText;
+  yetBtn.innerText = obj.contents;
+  li.id = obj.id;
+  li.append(yetBtn);
+  li.append(span);
+  li.append(delImg);
+  yetList.append(li); // dom 삽입
+  // del , done 처리
+
+  // 숫자 업데이트
+
+  delImg.addEventListener('click', del);
+  yetBtn.addEventListener('click', done);
+
+  updateNum();
+};
+
+// 함수4, render in doneList
+const setDone = (obj) => {
+  const li = document.createElement('li');
+  const span = document.createElement('span');
+  span.innerText = obj.contents;
+  li.id = obj.id;
+  li.append(span);
+  doneList.append(li);
+  updateNum();
+};
+
+// 함수 5, delete data both render and localStorage
+const del = (e) => {
+  // 1. delete rendering data in yet
+  e.target.parentElement.remove();
+
+  // 2. delete localStorage data in yet
+  const id = e.target.parentElement.id;
+  let temp = JSON.parse(localStorage.getItem(yetText));
+  temp = temp.filter((i) => i.id != id);
+  localStorage.setItem(yetText, JSON.stringify(temp));
+  updateNum();
+};
+// 함수 6, delete data both render and localStorage
+const done = (e) => {
+  const id = e.target.parentElement.id;
+  const text = e.target.parentElement.innerText.slice(4, -1);
+  // 1. delete render
+  e.target.parentElement.remove();
+
+  // 2.localStorage
+  // 2.1. delete yet in localStorage
+
+  const temp = JSON.parse(localStorage.getItem(yetText));
+  const yet = temp.filter((i) => i.id != id);
+  const done = temp.filter((i) => i.id == id);
```

Comment:
local storage에 `doneList`도 저장해 두셨는데 `yetList`만 가지고 필터링한 이유가 있으신가요? 진행 중인 할일 목록(yetList)에 있는 항목들을 기준으로 `yet`과 `done`을 나누기 때문에 완료한 할일 목록(doneList)의 개수가 제대로 카운팅되지 않는 것 같습니다.

또한 변수명이 list를 의미하는 건지, 그 안의 항목들을 의미하는 건지 약간 모호한 느낌도 있는 것 같아요. 아래처럼 수정하면 어떨까요? 또한 `iterator`의 이름도 명확히 작성하는 게 좋습니다!

```suggestion
  const temp = JSON.parse(localStorage.getItem(yetText));
  const yetList = temp.filter((todo) => todo.id != id);
  const doneList = temp.filter((todo) => todo.id == id);
```